src/ripple/ledger/impl/View.cpp

```diff
@@ -1216,41 +1021,35 @@ trustDelete (ApplyView& view,
                  beast::Journal j)
 {
     // Detect legacy dirs.
-    bool        bLowNode    = sleRippleState->isFieldPresent (sfLowNode);
-    bool        bHighNode   = sleRippleState->isFieldPresent (sfHighNode);
     std::uint64_t uLowNode    = sleRippleState->getFieldU64 (sfLowNode);
     std::uint64_t uHighNode   = sleRippleState->getFieldU64 (sfHighNode);
-    TER         terResult;
 
     JLOG (j.trace())
         << "trustDelete: Deleting ripple line: low";
-    terResult   = dirDelete(view,
-        false,
-        uLowNode,
-        keylet::ownerDir (uLowAccountID),
-        sleRippleState->key(),
-        false,
-        !bLowNode,
-        j);
 
-    if (tesSUCCESS == terResult)
+    if (view.dirRemove(
+            keylet::ownerDir(uLowAccountID),
+            uLowNode,
+            sleRippleState->key(),
+            false))
     {
         JLOG (j.trace())
                 << "trustDelete: Deleting ripple line: high";
-        terResult   = dirDelete (view,
-            false,
-            uHighNode,
-            keylet::ownerDir (uHighAccountID),
-            sleRippleState->key(),
-            false,
-            !bHighNode,
-            j);
+
+        if (!view.dirRemove(
+                keylet::ownerDir(uHighAccountID),
+                uHighNode,
+                sleRippleState->key(),
+                false))
+        {
+            return tefBAD_LEDGER;
+        }
     }
 
     JLOG (j.trace()) << "trustDelete: Deleting ripple line: state";
     view.erase(sleRippleState);
 
-    return terResult;
+    return tesSUCCESS;
```

Comment:
On second thought, this should return `tefBAD_LEDGER` if the first `dirRemove` fails.

I'm assuming this is a similar situation to what Mike and Scott were discussing
https://github.com/ripple/rippled/pull/2581/files/7e4b9d5a3d51bd083c38c586d219d5219d9c5d04#r196483659
Previously, this would always call `view.erase(sleRippleState)`. Currently, `erase` isn't called if the second `dirRemove` fails. It sounds like it should be fine to immediately return `tefBAD_LEDGER` after each `dirRemove` it fails.

Reply:
Right, I was thinking about this as well after reading that discussion last night. I agree with your assessment as well. I believe I've implemented what you were suggesting in https://github.com/JoeLoser/rippled/commit/f810d6742dc4139aeec09556bb5bd054ec0ea0da